test/jdk/java/util/zip/DeInflate.java

```diff
@@ -118,27 +118,55 @@ static void checkByteBufferReadonly(Deflater def, Inflater inf,
         } catch (ReadOnlyBufferException robe) {}
     }
 
-    static void check(Deflater def, byte[] in, int len,
-                      byte[] out1, byte[] out2, boolean nowrap)
+    /**
+     * Uses the {@code def} deflater to deflate the input data {@code in} of length {@code len}.
+     * A new {@link Inflater} is then created within this method to inflate the deflated data. The
+     * inflated data is then compared with the {@code in} to assert that it matches the original
+     * input data.
+     * This method repeats these checks for the different overloaded methods of
+     * {@code Deflater.deflate(...)} and {@code Inflater.inflate(...)}
+     *
+     * @param def    the deflater to use for deflating the contents in {@code in}
+     * @param in     the input content
+     * @param len    the length of the input content to use
+     * @param nowrap will be passed to the constructor of the {@code Inflater} used in this
+     *               method
+     * @throws Throwable if any error occurs during the check
+     */
+    static void check(Deflater def, byte[] in, int len, boolean nowrap)
         throws Throwable
     {
-        Arrays.fill(out1, (byte)0);
-        Arrays.fill(out2, (byte)0);
-
+        byte[] tempBuffer = new byte[len];
+        byte[] out1, out2;
+        int m = 0, n = 0;
+        Inflater inf = new Inflater(nowrap);
         def.setInput(in, 0, len);
         def.finish();
-        int m = def.deflate(out1);
 
-        Inflater inf = new Inflater(nowrap);
-        inf.setInput(out1, 0, m);
-        int n = inf.inflate(out2);
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(len)) {
+            while (!def.finished()) {
+                int temp_counter = def.deflate(tempBuffer);
+                m += temp_counter;
+                baos.write(tempBuffer, 0, temp_counter);
+            }
+            out1 = baos.toByteArray();
+            baos.reset();
 
-        if (n != len ||
-            !Arrays.equals(Arrays.copyOf(in, len), Arrays.copyOf(out2, len)) ||
-            inf.inflate(out2) != 0) {
-            System.out.printf("m=%d, n=%d, len=%d, eq=%b%n",
-                              m, n, len, Arrays.equals(in, out2));
-            throw new RuntimeException("De/inflater failed:" + def);
+            inf.setInput(out1, 0, m);
+
+            while (!inf.finished()) {
+                int temp_counter = inf.inflate(tempBuffer);
+                n += temp_counter;
+                baos.write(tempBuffer, 0, temp_counter);
+            }
+            out2 = baos.toByteArray();
+            if (n != len ||
+                !Arrays.equals(in, 0, len, out2, 0, len)) ||
```

Comment:
There's a `)` too much. you should at least run it a single time before pushing :)

Reply:
And while your on it, can you please also fix the other occurences of `Arrays.equals()`.

Reply:
oops, I made these change on Lpar and tested there & pushed the changes from local machine.

current changes are working fine.

really sorry.. :-(